scarb/src/ops/package.rs

```diff
@@ -88,10 +96,53 @@ fn before_package(ws: &Workspace<'_>) -> Result<()> {
     Ok(())
 }
 
-#[tracing::instrument(level = "trace", skip(_opts, ws))]
+#[derive(Serialize)]
+struct VcsInfo {
+    sha1: String,
+    path_in_vcs: String,
+}
+
+#[allow(clippy::dbg_macro)]
+fn extract_vcs_info(pkg: &Package, opts: &PackageOpts) -> Result<Option<VcsInfo>> {
+    if let Ok(repo) = gix::discover(pkg.root()) {
+        // There is nothing to parse if repository is empty, so HEAD reference needs to exist.
+        if let Ok(sha1) = repo.rev_parse_single("HEAD") {
+            // `git status -s` output is empty only if there are no changes at all, but always returns success.
+            // `git diff-index --quiet HEAD` returns status code correctly, but doesn't take untracked files into account.
+            let output = Command::new("git")
```

Comment:
Please try to reuse & extend the code used in `GitSource`, it tries to tame user's `git` installation, for example with this envs: https://github.com/software-mansion/scarb/blob/95885ad4913862c7c120cff4ca09a32d75f61bc0/scarb/src/sources/git/client.rs#L333-L347

Don't build `Command` here manually; instead let's maybe try to construct a `GitDatabase` object we have and call its method?